FluentValidation.AutoValidation.Mvc/src/Filters/FluentValidationAutoValidationActionFilter.cs

```diff
@@ -30,8 +33,7 @@ public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionE
             {
                 var actionDescriptor = context.ActionDescriptor;
 
-                // @todo figure out a better way to retrieve the attribute since using the `context.ActionDescriptor.EndpointMetadata` is not recommended for application code
-                if (autoValidationMvcConfiguration.ValidationStrategy == ValidationStrategy.Annotations && !actionDescriptor.EndpointMetadata.OfType<FluentValidationAutoValidationAttribute>().Any())
+                if (autoValidationMvcConfiguration.ValidationStrategy == ValidationStrategy.Annotations && !context.HttpContext.GetEndpoint().Metadata.OfType<FluentValidationAutoValidationAttribute>().Any())
```

Comment:
@alshuriga the `HttpContext.GetEndpoint()` method can return null. Can you move the endpoint to a new variable called `endpoint` above the `var actionDescriptor` and check for null in the if at line 36?